src/box/lua/config/applier/box_cfg.lua

```diff
@@ -283,11 +283,26 @@ local function apply(config)
     -- mix instances with data from different replicasets into one
     -- replicaset.
     local names = configdata:names()
-    box_cfg.instance_name = names.instance_name
-    box_cfg.replicaset_name = names.replicaset_name
     box_cfg.instance_uuid = names.instance_uuid
     box_cfg.replicaset_uuid = names.replicaset_uuid
 
+    -- Names are applied only if they're already in snap file.
+    -- Otherwise, master must apply names after box.cfg.
+    if not configdata._missing_names[names.instance_name] then
+        box_cfg.instance_name = names.instance_name
+    end
+    if not configdata._missing_names[names.replicaset_name] then
+        box_cfg.replicaset_name = names.replicaset_name
+    end
+
+    if is_startup then
+        for name, uuid in pairs(configdata._missing_names) do
+            local warning = string.format('box_cfg.apply: name %s for '..
+                                          '%s is missing', name, uuid)
```

Comment:
Nit: Is missing where? What should I do?

As for me, this message is not clear for a user.

Reply:
Is missing from snapshot. I changed the msg. The user should do nothing, it'll be automatically applied.